src/Compilers/CSharp/Portable/Binder/Binder_Patterns.cs

```diff
@@ -33,7 +35,7 @@ private BoundExpression BindIsPatternExpression(IsPatternExpressionSyntax node,
             }
 
             uint inputValEscape = GetValEscape(expression, LocalScopeDepth);
-            BoundPattern pattern = BindPattern(node.Pattern, expression.Type, inputValEscape, permitDesignations: true, hasErrors, diagnostics);
+            BoundPattern pattern = BindPattern(node.Pattern, expression.Type!, inputValEscape, permitDesignations: true, hasErrors, diagnostics);
```

Comment:
>expression.Type! [](start = 61, length = 16)

I see a path where `expression.Type` is null getting here

Reply:
I don't.  The previous if statement ensure it isn't.  `BadExpression` returns an expression whose type is never null.

---
In reply to: [380880989](https://github.com/dotnet/roslyn/pull/41503#discussion_r380880989) [](ancestors = 380880989)